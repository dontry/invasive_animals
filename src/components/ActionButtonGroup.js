import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Button from "material-ui/Button";

const styles = {
  button: {
    marginLeft: 5,
    marginRight: 5
  }
};

const ActionButton = props => (
  <Button
    variant={props.raised ? "raised" : "flat"}
    style={styles.button}
    className={props.className}
    color="primary"
    disabled={props.disabled}
    onClick={props.action}
    type={props.type}
  >
    {props.label}
  </Button>
);

const ActionButtonGroup = ({ className, primaryProps, secondaryProps }) => {
  return (
    <div className={className}>
      <ActionButton {...primaryProps} />
      <ActionButton {...secondaryProps} />
    </div>
  );
};

ActionButtonGroup.propTypes = {
  className: PropTypes.string,
  primaryProps: PropTypes.shape({
    className: PropTypes.string,
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.string
  }),
  secondaryProps: PropTypes.shape({
    className: PropTypes.string,
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.string
  })
};

ActionButtonGroup.defaultProps = {
  className: "",
  primaryProps: {
    className: "",
    raised: true,
    label: "label",
    primary: false,
    action() {
      console.log("onClick");
    },
    disabled: true,
    type: "button"
  },
  secondaryProps: {
    className: "",
    label: "label",
    raised: false,
    primary: false,
    action() {
      console.log("onClick");
    },
    disabled: true,
    type: "button"
  }
};

export default withStyles(styles)(ActionButtonGroup);
