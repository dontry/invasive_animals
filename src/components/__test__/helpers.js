import React from "react";
import defaultTheme from "../../assets/theme";
import { ThemeProvider } from "styled-components";

export function shallowWithTheme(component, theme = defaultTheme) {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return shallow(component, { context });
}

export function mountWithTheme(component, theme = defaultTheme) {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes // Needed so child components receive the theme propd
  });
}
