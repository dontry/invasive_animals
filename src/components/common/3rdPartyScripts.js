import React from "react";
import Script from "react-load-script";

export const TableauScript = () => {
  return (
    <Script
      url="https://public.tableau.com/javascripts/api/viz_v1.js"
      onCreate={() => {
        console.log("create script");
      }}
      onError={() => {
        console.log("load script error");
      }}
      onLoad={() => {
        console.log("load tableau script");
      }}
    />
  );
};
