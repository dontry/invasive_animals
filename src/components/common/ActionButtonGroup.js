import React from "react";
import PropTypes from "prop-types";
import Button from "material-ui/Button";
import styled from "styled-components";

const ButtonGroupWrapper = styled.div`
  text-align: center;
`;

export const StyledButton = styled(Button)`
  && {
    margin-left: 5px;
    margin-right: 5px;
    background-color: ${props =>
      props.theme.palette
        ? props.type
          ? props.trait
            ? props.theme.palette[props.type][props.trait]
            : props.theme.palette[props.type].main
          : props.theme.primary.main
        : "#fff"};
    color: ${props => props.theme.palette ? props.theme.palette[props.type].contrastText : "#222"};
    font-size: ${props => props.theme.palette ? props.theme.text_size.size : "1em"};
    height: ${props => props.height || "auto"};
    width: ${props => props.width || "auto"};
    padding ${props => props.padding || "auto"};

    &:hover {
      background-color: ${props => props.theme.palette ? props.theme.palette[props.type].dark : "#acacac"};
    }
  }
`;

export const ActionButton = ({
  raised,
  action,
  label,
  height,
  width,
  padding,
  type = "primary",
  trait = "main"
}) => (
  <StyledButton
    variant={raised ? "raised" : "flat"}
    onClick={action}
    type={type}
    trait={trait}
    height={height}
    width={width}
    padding={padding}
  >
    {label}
  </StyledButton>
);

const ActionButtonGroup = ({ btnStyle, primaryProps, secondaryProps }) => {
  return (
    <ButtonGroupWrapper btnStyle={btnStyle}>
      <ActionButton className="action-btn" {...primaryProps} />
      <ActionButton className="action-btn" {...secondaryProps} />
    </ButtonGroupWrapper>
  );
};

ActionButtonGroup.propTypes = {
  style: PropTypes.object.isRequired,
  primaryProps: PropTypes.shape({
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.string
  }),
  secondaryProps: PropTypes.shape({
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.string
  })
};

ActionButtonGroup.defaultProps = {
  style: {},
  primaryProps: {
    raised: true,
    label: "Primary",
    primary: false,
    action() {
      console.log("onClick");
    },
    disabled: false,
    type: "primary",
    trait: "main"
  },
  secondaryProps: {
    label: "Secondary",
    raised: false,
    primary: false,
    action() {
      console.log("onClick");
    },
    disabled: false,
    type: "secondary",
    trait: "light"
  }
};

export default ActionButtonGroup;
