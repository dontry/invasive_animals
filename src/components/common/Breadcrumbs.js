import React, { Children } from "react";
import PropTypes from "prop-types";
import { grey } from "material-ui/colors";
import styled from "styled-components";
import { Link } from "react-router-dom";
import NavigationLink from "./NavLink";
import Icon from "material-ui/Icon";
import { findRouteName } from "../../utils/routing";

export const Separator = styled.span`
  display: inline-block;
  color: ${props => props.color || grey[400]};
  text-align: center;
  flex-shrink: 0;
  padding: ${props => props.padding || "0.5rem"};
`;

Separator.displayName = "Separator";

export const BreadcrumbsItem = ({match,  hasSeparator = false }) => {
  let routeName = findRouteName(match.url);
  let content = null;
  if (routeName) {
    routeName = routeName !== "HOME" ? routeName : <Icon style={{ verticalAlign: "-0.2rem" }}>home</Icon> 
    return (
      <span>
        <NavigationLink textSize="small" to={match.url || ''}>{routeName}</NavigationLink>
        {hasSeparator && <Separator>/</Separator>}
      </span>
    );
  }
};

BreadcrumbsItem.propTypes = {
  hasSeparator: PropTypes.bool
};

//Atlakit breadcrumb
const { toArray } = Children;

export const Breadcrumbs = ({ children }) => {
  const allNonEmptyItems = toArray(children);
  return allNonEmptyItems.map((child, index) =>
    React.cloneElement(child, {
      hasSeparator: index < children.length - 1
    })
  );
};
