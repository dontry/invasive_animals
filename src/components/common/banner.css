.slick-dots {
    bottom: 25px;
}


.slick-dots li button::before {
    font-size: 12px;
    color: rgba(217, 243, 186, 1);
}

.slick-dots li.slick-active button::before {
    color: rgb(139, 195, 74);
}