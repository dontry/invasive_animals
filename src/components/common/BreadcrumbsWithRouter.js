import React, { Component } from "react";
import { Route, withRouter } from "react-router-dom";
import { Breadcrumbs, BreadcrumbsItem, Separator } from "./Breadcrumbs";

function renderBreadcrumbItems(paths) {
  return paths.map((p, index) => {
    if (index < paths.length - 1) {
      return (
        <span>
          <Route key={p} path={p} component={BreadcrumbsItem} />
          <Separator>/</Separator>
        </span>
      );
    } else {
      return <Route key={p} path={p} component={BreadcrumbsItem} />;
    }
  });
}

const BreadcrumbsWithRouter = ({ location, match }) => {
  const paths = [];
  location.pathname.split("/").reduce((prev, curr, index) => {
    paths[index] = `${prev}/${curr}`;
    return paths[index];
  });
  if(paths[1] !== "/") paths[0] = "/";
  const breadcrumbsItems = renderBreadcrumbItems(paths);
  return <Breadcrumbs>{breadcrumbsItems}</Breadcrumbs>;
};

export default withRouter(BreadcrumbsWithRouter);
