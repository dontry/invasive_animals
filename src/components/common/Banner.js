import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
//Material UI
import Grid from "material-ui/Grid";

import Slider from "react-slick";
import "./banner.css";

const StyledSlider = styled(Slider)`
`;

const BannerWrapper = styled.div`
  position: relative;
  box-sizing: border-box;
  min-height: 300px;
  height: 45vh;
  width: 100%;
  z-index: 1;
  &::before {
    content: "";
    position: absolute;
    margin: ${props => (props.padding ? "-" + props.padding : "-1rem")};
  }
  @media screen and (max-width: 400px) {
    & {
      height: 30vh;
    }
  }
`;

const BackgroundImage = styled.img`
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: -10;
  filter: brightness(80%);
`;

const BlurbWrapper = styled(Grid)`
  && {
    padding-top: 1rem;
    padding-left: 2rem;
  }
`;

const Banner = ({ timeout = 0, heading, description, banners, textColor }) => {
  const settings = {
    dots: true,
    infinite: true,
    lazyLoad: true,
    fade: true,
    autoplay: true,
    speed: 5000,
    slidesToShow: 1,
    slidesToScroll: 1,
    arrows: false
  };
  return (
    <StyledSlider {...settings}>
      {banners.map((banner, index) => (
        <BannerWrapper key={`banner-${index}`}>
          <BackgroundImage src={banner.image} />
          <BlurbWrapper
            container
            direction="column"
            justify="center"
            alignItems="flex-start"
          >
            <Grid item xs={12} md={9}>
              {banner.description}
            </Grid>
          </BlurbWrapper>
        </BannerWrapper>
      ))}
    </StyledSlider>
  );
};

Banner.propTypes = {
  heading: PropTypes.string.isRequired,
  subheading: PropTypes.string,
  description: PropTypes.string,
  imgUrl: PropTypes.string
};

Banner.defaultProps = {
  heading: "",
  description: "",
  imgUrl: "",
  textColor: "#fff"
};

export default Banner;
