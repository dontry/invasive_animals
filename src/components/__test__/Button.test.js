import React from "react";
import renderer from "react-test-renderer";
import { ActionButton } from "../common/ActionButtonGroup";
import { shallowWithTheme } from "./helpers";

describe("<ActionButton />", () => {
  it("ActionButton default props", () => {
    const wrapper = mount(<ActionButton label="button" />);
    const json = toJson(wrapper);
    expect(wrapper.props().type).toBe("primary");
    expect(wrapper.props().trait).toBe("main");
    expect(json).toMatchSnapshot();
  });
});
