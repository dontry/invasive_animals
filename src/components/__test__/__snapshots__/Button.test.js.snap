// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> ActionButton default props 1`] = `
.c0.c0 {
  margin-left: 5px;
  margin-right: 5px;
  background-color: #fff;
  color: #222;
  font-size: 1em;
  height: auto;
  width: auto;
  padding: auto;
}

.c0.c0:hover {
  background-color: #acacac;
}

<ActionButton
  label="button"
>
  <Styled(WithStyles(Button))
    trait="main"
    type="primary"
    variant="flat"
  >
    <WithStyles(Button)
      className="c0"
      trait="main"
      type="primary"
      variant="flat"
    >
      <Button
        className="c0"
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit-5",
            "disabled": "MuiButton-disabled-10",
            "fab": "MuiButton-fab-11",
            "flatPrimary": "MuiButton-flatPrimary-3",
            "flatSecondary": "MuiButton-flatSecondary-4",
            "focusVisible": "MuiButton-focusVisible-9",
            "fullWidth": "MuiButton-fullWidth-15",
            "label": "MuiButton-label-2",
            "mini": "MuiButton-mini-12",
            "raised": "MuiButton-raised-6",
            "raisedPrimary": "MuiButton-raisedPrimary-7",
            "raisedSecondary": "MuiButton-raisedSecondary-8",
            "root": "MuiButton-root-1",
            "sizeLarge": "MuiButton-sizeLarge-14",
            "sizeSmall": "MuiButton-sizeSmall-13",
          }
        }
        color="default"
        disableFocusRipple={false}
        disabled={false}
        fullWidth={false}
        mini={false}
        size="medium"
        trait="main"
        type="primary"
        variant="flat"
      >
        <WithStyles(ButtonBase)
          className="MuiButton-root-1 c0"
          disabled={false}
          focusRipple={true}
          focusVisibleClassName="MuiButton-focusVisible-9"
          trait="main"
          type="primary"
        >
          <ButtonBase
            centerRipple={false}
            className="MuiButton-root-1 c0"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-17",
                "focusVisible": "MuiButtonBase-focusVisible-18",
                "root": "MuiButtonBase-root-16",
              }
            }
            disableRipple={false}
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-9"
            tabIndex="0"
            trait="main"
            type="primary"
          >
            <button
              className="MuiButtonBase-root-16 MuiButton-root-1 c0"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              trait="main"
              type="primary"
            >
              <span
                className="MuiButton-label-2"
              >
                button
              </span>
              <WithStyles(TouchRipple)
                center={false}
                innerRef={[Function]}
              >
                <TouchRipple
                  center={false}
                  classes={
                    Object {
                      "child": "MuiTouchRipple-child-23",
                      "childLeaving": "MuiTouchRipple-childLeaving-24",
                      "childPulsate": "MuiTouchRipple-childPulsate-25",
                      "ripple": "MuiTouchRipple-ripple-20",
                      "ripplePulsate": "MuiTouchRipple-ripplePulsate-22",
                      "rippleVisible": "MuiTouchRipple-rippleVisible-21",
                      "root": "MuiTouchRipple-root-19",
                    }
                  }
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    className="MuiTouchRipple-root-19"
                    component="span"
                    enter={true}
                    exit={true}
                  >
                    <span
                      className="MuiTouchRipple-root-19"
                    />
                  </TransitionGroup>
                </TouchRipple>
              </WithStyles(TouchRipple)>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </Button>
    </WithStyles(Button)>
  </Styled(WithStyles(Button))>
</ActionButton>
`;
